2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+- What to say:
+    - TODO
+- Notes:
+    - Mention, that practical part will focuse on ATF and OPTEE?
+
+---
+
+# What to integrate and why?
+
+
+.center[ <img src="./img/TEE_ARM_Cortex-a_with_SM_and_apps.svg"> ]
+
+.center[<span style="font-size: 30px;">
+            A bunch of stuff to integrate, huh?
+        </span>]
+
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To decompose what parts should be integrated to use
+  TEE on ARM.
+- What to say:
+    - TODO
+- Notes:
+    - What to integrate:
+        - ATF (ARM Trusted Firmware);
+         - OP-TEE OS;
+        - TAs (like crypto or PKCS#11);
+        - Userspace clients (like OpenSSL), libs (at least for communicating
+          with TEE) and tools (TEE-specific tools, like tee-supplicant for
+          OP-TEE);
+        - Linux Kernel drivers and devicetree configuration.
+    - Mention U-Boot integration as well.
+
+---
+
+# Compilation from source
+
+.left-code-column50[
+```bash
+meta-arm
+(...)
+├── recipes-bsp
+(...)
+│   ├── trusted-firmware-a
+(...)
+│   │   ├── tf-a-tests_2.10.0.bb
+│   │   ├── trusted-firmware-a_2.10.3.bb
+│   │   └── trusted-firmware-a.inc
+│   ├── trusted-firmware-m
+│   │   ├── trusted-firmware-m_2.0.0.bb
+│   │   ├── trusted-firmware-m-2.0.0-src.inc
+│   │   ├── trusted-firmware-m.inc
+(...)
+├── recipes-security
+│   ├── optee
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+│   │   ├── optee-client_4.1.0.bb
+│   │   ├── optee-client.inc
+│   │   ├── optee-examples_4.1.0.bb
+│   │   ├── optee.inc
+(...)
+│   │   ├── optee-os_4.1.0.bb
+│   │   ├── optee-os.inc
+(...)
+```
+]
+.right-code-column50[
+```bash
+meta-arm-bsp
+(...)
+├── recipes-bsp
+(...)
+│   ├── trusted-firmware-a
+(...)
+│   │   ├── fiptool-native_2.8.6.bb
+│   │   ├── tf-a-tests_2.8.0.bb
+│   │   ├── tf-a-tests_%.bbappend
+│   │   ├── trusted-firmware-a_%.bbappend
+(...)
+│   ├── trusted-firmware-m
+(...)
+│   │   ├── trusted-firmware-m_%.bbappend
+│   │   └── trusted-firmware-m-corstone1000.inc
+(...)
+├── recipes-security
+│   ├── optee
+(...)
+│   │   ├── optee-client
+│   │   │   ├── tee-supplicant@.service
+│   │   │   └── tee-supplicant.sh
+(...)
+│   │   ├── optee-ftpm_%.bbappend
+│   │   ├── optee-os_4.%.bbappend
+(...)
+```
+]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To show that meta-arm has everithing neccessary and
+  even more.
+- What to say:
+    - TODO
+- Notes:
+    - Porting problems:
+        - TODO
+
+---
+
+# Using vendor binaries
+
+.center[ <img src="./img/vendor-binaries-meme.png" height="350px"> ]
+
+.center[ <span style="font-size:30px;">There's some kind of mess here!</span>]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: To say that though vendor supplied binaries were
+  sopposed to make development easier, they made it even worse.
+- What to say:
+    - TODO
+- Notes:
+    - Weak points:
+        - You have to trust vendor binaries;
```

Comment:
Is there a way to not use them? What could be done about it?

Reply:
Compilation from source, I will say it during presentation without stating it on slides explicitely.

Reply:
Do all vendors provide the source code to build them? What are the things the engineer needs to do to check if they can implement a TEE in their project or on a certain board?

Reply:
Yes you're absolutely right